### ISSUE TITLE
Issues with text reflow and proposed solutions

### ISSUE THREAD

**Issue by thousandsofthem** [kiwibrowser/android#86]
Issues with text reflow and proposed solutions.

# Unnecessary big right/left margins

* Screenshot: Opera browser
![Screenshot_20190701-013325_Opera beta](https://user-images.githubusercontent.com/736279/60403069-2425ae80-9ba1-11e9-9905-461d34660a1d.jpg)

* Screenshot: Kiwi Genie
![Screenshot_20190701-013528_Kiwi Browser](https://user-images.githubusercontent.com/736279/60403132-04db5100-9ba2-11e9-88d3-fc9db1ca0389.jpg)

Proposed solution: reduce added margins

# Movement issues

When text reflow is turned on, browser does not allow free moving view area on a page. Then trying to move page left/right it bounces back. Then moving up/down it affects scrolling speed, scrolling becomes faster/slower/jittery 
Example:
http://lon-screenshots.s3.amazonaws.com/2019-07-01_01_48_41_2019-01-07_01-37-54.mp4

Proposed solution:

* Do not affect scrolling anyhow
* Do not apply anything regarding text reflow while user interacts with a page, scrolling in progress etc
* Adjust text width on per block basis, see how Opera does it, works quote great. Check out e.g. on deeply nested discussions on news.ycombinator.com
* Whole reflowing thing need to be done _once_ on zoom in/out. Again, see how Opera does it

Page on screenshots: https://gist.github.com/rjhansen/67ab921ffb4084c865b3618d6955275f
OS: Android 8.0 stock version by Samsung
Device: Samsung S8+

**Comment by timunie** [comment]
I really love this browser and it will be my entire replacement, if this feature is less buggy (I know that this is a free time project and i don't want to hurry you :+1: ). 

Just in case it helps you with the development: This feature doesn't work at all on this page [https://www.formel1.de/forum/viewtopic.php?t=62076](https://www.formel1.de/forum/viewtopic.php?t=62076). it works in Opera quite good. 

I would make a PR if I had any idea how to fix it, but I don't have :disappointed: 


Happy coding
Tim

**Comment by gggirlgeek** [comment]
Yes please! Reddit is another example. Almost impossible to deal with in long threads.

**Comment by djp101** [comment]
This is still an issue on version 107. Makes it's unusable on many documentation websites.
